server/update_branch.go

```diff
@@ -0,0 +1,45 @@
+package server
+
+import (
+	"context"
+	"strings"
+
+	"github.com/mattermost/mattermost-server/mlog"
+
+	"github.com/google/go-github/v28/github"
+)
+
+func (s *Server) handleUpdateBranch(eventIssueComment IssueComment) {
+	client := NewGithubClient(s.Config.GithubAccessToken)
+	prGitHub, _, err := client.PullRequests.Get(context.Background(), *eventIssueComment.Repository.Owner.Login, *eventIssueComment.Repository.Name, *eventIssueComment.Issue.Number)
+	if err != nil {
+		mlog.Error("Error getting the latest PR information from github", mlog.Err(err))
+		return
+	}
+	pr, err := s.GetPullRequestFromGithub(prGitHub)
+	if err != nil {
+		mlog.Error("Error Updating the PR in the DB", mlog.Err(err))
+		return
+	}
+
+	userComment := *eventIssueComment.Comment.User
+	if !s.checkUserPermission(userComment.GetLogin(), pr.RepoOwner) {
+		s.sendGitHubComment(pr.RepoOwner, pr.RepoName, pr.Number, "Looks like you dont have permissions to trigger this command.\n Only available for Org members")
+		return
+	}
+
+	opt := &github.PullReqestBranchUpdateOptions{
+		ExpectedHeadSHA: github.String(pr.Sha),
+	}
+
+	_, resp, err := client.PullRequests.UpdateBranch(context.Background(), pr.RepoOwner, pr.RepoName, pr.Number, opt)
```

Comment:
Does this work for forks?

Reply:
github api does not say that dont work for forks, i think it will